File: sdkwork-ai-sdk-flutter/lib/src/api/context.dart
import '../http/client.dart';
import '../models.dart';

class ContextApi {
  final HttpClient _client;
  
  ContextApi(this._client);

  /// Chat with context
  Future<ContextResponse?> chatWith(String context_id, ChatWithContextRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/context/${context_id}/chat/completions'), body: body, contentType: 'application/json');
    return response is ContextResponse ? response : null;
  }

  /// Create context
  Future<ContextResponse?> createContext(ContextCreateRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/context/create'), body: body, contentType: 'application/json');
    return response is ContextResponse ? response : null;
  }

  /// Get context
  Future<ContextResponse?> getContext(String context_id) async {
    final response = await _client.get(ApiPaths.aiPath('/context/${context_id}'));
    return response is ContextResponse ? response : null;
  }

  /// Delete context
  Future<void> deleteContext(String context_id) async {
    await _client.delete(ApiPaths.aiPath('/context/${context_id}'));
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/context.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import ChatWithContextRequest, ContextCreateRequest, ContextResponse

class ContextApi:
    """context API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def chat_with(self, context_id: str, body: ChatWithContextRequest) -> ContextResponse:
        """Chat with context"""
        return self._client.post(f"/ai/v3/context/{context_id}/chat/completions", json=body)

    def create_context(self, body: ContextCreateRequest) -> ContextResponse:
        """Create context"""
        return self._client.post(f"/ai/v3/context/create", json=body)

    def get_context(self, context_id: str) -> ContextResponse:
        """Get context"""
        return self._client.get(f"/ai/v3/context/{context_id}")

    def delete_context(self, context_id: str) -> None:
        """Delete context"""
        return self._client.delete(f"/ai/v3/context/{context_id}")
